File: src/main/java/um/haberes/core/model/view/AsignadoCategoria.java
package um.haberes.core.model.view;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.annotations.Immutable;

import java.math.BigDecimal;

@Entity
@Table(
        name = "vw_asignado_categoria",
        uniqueConstraints = {@UniqueConstraint(columnNames = {"legajoId", "dependenciaId", "categoriaId"})}
)
@Immutable
@Getter
@Setter
@EqualsAndHashCode
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class AsignadoCategoria {

    @Id
    private String uniqueId = null;

    private Long legajoId = null;

    private Integer dependenciaId = null;

    private Integer categoriaId = null;

    private Long periodoDesde = null;

    private Long periodoHasta = null;

    private BigDecimal basicoDesde = BigDecimal.ZERO;

    private BigDecimal basicoHasta = BigDecimal.ZERO;
}

File: src/main/java/um/haberes/core/kotlin/model/view/AsignadoCategoria.kt
package um.haberes.core.kotlin.model.view

import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table
import jakarta.persistence.UniqueConstraint
import org.hibernate.annotations.Immutable
import java.math.BigDecimal

@Entity
@Table(
    name = "vw_asignado_categoria",
    uniqueConstraints = [UniqueConstraint(columnNames = ["legajoId", "dependenciaId", "categoriaId"])]
)
@Immutable
data class AsignadoCategoria(

    @Id
    var uniqueId: String? = null,
    var legajoId: Long? = null,
    var dependenciaId: Int? = null,
    var categoriaId: Int? = null,
    var periodoDesde: Long? = null,
    var periodoHasta: Long? = null,
    var basicoDesde: BigDecimal = BigDecimal.ZERO,
    var basicoHasta: BigDecimal = BigDecimal.ZERO

)
